proxy/http/HttpSM.cc

```diff
@@ -3405,6 +3657,66 @@ HttpSM::tunnel_handler_ua(int event, HttpTunnelConsumer *c)
   return 0;
 }
 
+int
+HttpSM::tunnel_handler_trailer_ua(int event, HttpTunnelConsumer *c)
```

Comment:
Is it possible to add support for trailer before or after H2 to Origin separately? I guess it was added for gRPC, but I assume it'll be used for H1 as well. And if we support trailer header, that should be explicitly stated on release note.

Reply:
Could separate it out.  What is present now has not been tested for a while (AFAIK).  I did very basic tests early on with gRPC.

The goal for the support is gRPC which is only needed for HTTP/2.  We will need that support very soon after the base H2 to origin is committed.  

I agree that it would be nice to support for H1 as well, but do we have a use case for trailers with H1?

Reply:
Just to wrap up this thread, we can work on trailer/gRPC further in a separate PR. Thank you, @maskit and @shinrich, for clarifying this.